bridge-history-api/internal/controller/api/history_controller.go

```diff
@@ -0,0 +1,101 @@
+package api
+
+import (
+	"errors"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-redis/redis/v8"
+	"gorm.io/gorm"
+
+	"scroll-tech/bridge-history-api/internal/logic"
+	"scroll-tech/bridge-history-api/internal/types"
+)
+
+// HistoryController contains the query claimable txs service
+type HistoryController struct {
+	historyLogic *logic.HistoryLogic
+}
+
+// NewHistoryController return HistoryController instance
+func NewHistoryController(db *gorm.DB, redis *redis.Client) *HistoryController {
+	return &HistoryController{
+		historyLogic: logic.NewHistoryLogic(db, redis),
+	}
+}
+
+// GetL2ClaimableWithdrawalsByAddress defines the http get method behavior
+func (c *HistoryController) GetL2ClaimableWithdrawalsByAddress(ctx *gin.Context) {
+	var req types.QueryByAddressRequest
+	if err := ctx.ShouldBind(&req); err != nil {
+		types.RenderFailure(ctx, types.ErrParameterInvalidNo, err)
+		return
+	}
+
```

Comment:
add some check with pagesize. Maximum is 100 or 50

Reply:
type Form struct {
   Age int `binding:"required,numeric,min=18,max=40"`
}

https://github.com/gin-gonic/gin/issues/1093

Reply:
That makes sense, I will set it as 100, and test it later.